### ISSUE TITLE
Move the functionality of `aiida.cmdline.utils.common.get_database_summary` to `StorageBackend.get_info`

### ISSUE THREAD

**Issue by sphuber** [aiidateam/aiida-core#5384]
The `get_database_summary` is currently used in `verdi storage info` and `verdi archive inspect`. Both of these essentially give an overview of the storage contents. For historical reason, the functionality of getting the info was split in the database and repository, but these are now unified in the `StorageBackend`. The functionality of `get_database_summary` should therefore be moved into `StorageBackend.get_info`.

Note that `get_database_summary` currently just uses the ORM and so is technically already independent of the `StorageBackend`, but there might still be an advantage to put it in `StorageBackend.get_info` anyway, because certain backends may have more efficient ways of retrieving the information other than going through the ORM which might be adding unnecessary overhead.

**Comment by chrisjsewell** [comment]
I would be more specific here and say, after #5331,
[`aiida/cmdline/commands/cmd_storage.py::storage_info`](https://github.com/aiidateam/aiida-core/pull/5331/files#diff-f3d13a2ae78add210c0cfb8765a16c3de937fd3fe0cac54a35a3d23928e9b953) and `PsqlDosBackend.get_info` will essentially be broke,
since they both treat `get_info` as if it is only returning information about the repository, contrary to the `get_info` docstring: "Return general information on the storage."

**Comment by ramirezfranciscof** [comment]
In the end the docstring is correct and the method was adapted, so we can remove the bug tag. But thanks for the "specificity" and the preemptive concern, I guess.

**Comment by ramirezfranciscof** [comment]
> I'll take care of moving the database info logic in another PR.

(quote from the [original PR](https://github.com/aiidateam/aiida-core/pull/5331#pullrequestreview-891019058))

@sphuber anyways this is not a priority right now for 2.0 no? Since this is mostly internal re-organization.

**Comment by sphuber** [comment]
It is not, but it is easy enough that I will do it soon anyway I think

**Comment by ramirezfranciscof** [comment]
Ok, I still think there might be some merit in discussing how to balance the prospect of having this command output more "unpredictable" information specific to the backend and more "reliable" and backend agnostic statistics.

For example, the current stats that the summary outputs are the aiida nodes, which is a general concept that is independent of the backend, so I don't think it is a bad idea to also get this information in an agnostic way (even though, as you mentioned elsewhere, it might not be the most ideally performant way).

**Comment by sphuber** [comment]
@ramirezfranciscof makes sense. See #5387 for a suggestion. I think this possibly addresses your concerns.
